### PR TITLE
refactoring GitSavvy settings and project-wise settings

### PR DESCRIPTION
**Pull request by randy3k**
this adds support of project-wise settings by refactoring settings related code.

A [new command](https://github.com/divmain/GitSavvy/blob/10b0ef94844eb823930be697a6fda659479e3e67/docs/README.md#editing-settings) `Preference: GitSavvy Project Settings` is also added to allow easier editing.

### CODE REVIEW

File: common/global_events.py [hunk 1]
```diff
@@ -61,3 +61,28 @@ class GsEditSettingsCommand(WindowCommand):
     """
     def run(self, **kwargs):
         self.window.run_command("edit_settings", kwargs)
+
```
Comment (stoivo) [inline common/global_events.py]: This should have been in the later commit but I ¯\_(ツ)_/¯

**Comment by stoivo** [comment]
Looking good

**Comment by randy3k** [comment]
Though I have been using this branch for a while, I am still not 100% nothing breaks. However, let's merge it and test.

File: common/global_events.py (hunk 1 above)
Comment (randy3k) [reply common/global_events.py]: Ya, maybe.

**Pull request merged**